guide/yaml/winrm/client.md

```diff
@@ -0,0 +1,125 @@
+
+## WinRM Connectivity Diagnostics
+
+If you are experiencing problems with a windows blueprint against a jclouds location 
+where Apache Brooklyn complains about failing to connect to the IP you should check those things.
+
+1. Apache Brooklyn is using correct username and password
+1. Apache Brooklyn can reach the IP of the provisioned machine. WinRM port 5985 or 5986 is also reachable from Apache Brooklyn.
+1. Check whether `WinRmMachineLocation#getDefaultUserMetadataString(ConfigurationSupportInternal)` is applied on the VM.
+   This script should be passed to the cloud and executed in order to configure WinRM according to Apache Brooklyn requirements for authentication.
+   So far windows startup script are known to be supported on AWS EC2 and VCloud Director.
+   If your cloud doesn't use this script then tune WinRM parameters accordingly.
+1. Check whether you use winrm over http or over https.
+  1. If you are using WinRM over http then make sure WinRM service on target VM has `AllowUnencrypted = true`
+
+If the quick list above doesn't help then follow the steps bellow.
+
+To speed up diagnosing the problem we advice to trigger a deployment with the JcloudsLocation flag `destroyOnFailure: false` so you can check status of the provisioned machine
+or try later different WinRM parameters with a Apache Brooklyn [BYON Location](../../ops/locations/index.html#byon).
+
+After you determined what is the username and the password you can proceed with next steps.
+*(Notice that for cloud providers which use Auto Generated password will not be logged.
+For these cases use Java Debug to retrieve ot or provision a VM manually with the same parameters when using Apache Brooklyn to provision a jclouds location.)*
+
+The first step is to find what is the winrm service configuration on the target host.
+
+1. If you have RDP access or KVM like access to the VM then check the winrm service status with the command bellow.
+   `winrm get winrm/config/service`
+   If you are using http you should have AllowUnencrypted to false.
+   Encryption is supported only over https.
+   Sample output:
+
+        MaxConcurrentOperations = 4294967295
+        MaxConcurrentOperationsPerUser = 1500
+        EnumerationTimeoutms = 240000
+        MaxConnections = 300
+        MaxPacketRetrievalTimeSeconds = 120
+        AllowUnencrypted = true
+        Auth
+            Basic = false
+            Kerberos = true
+            Negotiate = true
+            Certificate = false
+            CredSSP = true
+            CbtHardeningLevel = Relaxed
+        DefaultPorts
+            HTTP = 5985
+            HTTPS = 5986
+        IPv4Filter = *
+        IPv6Filter = *
+        EnableCompatibilityHttpListener = false
+        EnableCompatibilityHttpsListener = false
+        CertificateThumbprint
+        AllowRemoteAccess = true
+
+Use an Apache Brooklyn BYON blueprint to try easily other connection options.
+
+    location:
```

Comment:
I may add that they users test commands preliminary with BYON.
Such as administrative commands which require CredSSP